kvserver: nil pointer dereference in TestMergeQueue

**Describe the problem**

See this [build](https://teamcity.cockroachdb.com/viewLog.html?buildId=2841409&buildTypeId=Cockroach_UnitTests) run against the 21.1 branch.

```

------- Stdout: -------
=== RUN   TestMergeQueue/sticky-bit
    --- FAIL: TestMergeQueue/sticky-bit (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c1a2e7]

goroutine 196713 [running]:
testing.tRunner.func1.1(0x4182cc0, 0x737a9c0)
	/usr/local/go/src/testing/testing.go:1072 +0x30d
testing.tRunner.func1(0xc001b04f00)
	/usr/local/go/src/testing/testing.go:1075 +0x41a
panic(0x4182cc0, 0x737a9c0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).SetZoneConfig(0x0, 0xc008ca0fc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica.go:689 +0x47
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestMergeQueue.func5(0xc005694700, 0xc0056946f0, 0xc0056946f8, 0x0, 0xc0056946e0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:4144 +0xae
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestMergeQueue.func6(0xc001b04f00)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:4155 +0x302
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestMergeQueue.func12(0xc001b04f00)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:4234 +0x8f
testing.tRunner(0xc001b04f00, 0xc00539cf80)
	/usr/local/go/src/testing/testing.go:1123 +0xef
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1168 +0x2b3
```

+cc @tbg for routing.

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -4091,7 +4091,6 @@ func TestMergeQueue(t *testing.T) {
 	zoneConfig.RangeMinBytes = &rangeMinBytes
 	settings := cluster.MakeTestingClusterSettings()
 	sv := &settings.SV
-	kvserverbase.MergeQueueEnabled.Override(ctx, sv, true)
 	kvserver.MergeQueueInterval.Override(ctx, sv, 0) // process greedily
 
 	tc := testcluster.StartTestCluster(t, 2,
@@ -4114,7 +4113,8 @@ func TestMergeQueue(t *testing.T) {
 	store := tc.GetFirstStoreFromServer(t, 0)
 	// The cluster with manual replication disables the merge queue,
 	// so we need to re-enable.
-	kvserverbase.MergeQueueEnabled.Override(ctx, sv, true)
+	_, err := tc.ServerConn(0).Exec(`SET CLUSTER SETTING kv.range_merge.queue_enabled = true`)
+	require.NoError(t, err)
 	store.SetMergeQueueActive(true)
 
 	split := func(t *testing.T, key roachpb.Key, expirationTime hlc.Timestamp) {
@@ -4228,22 +4228,14 @@ func TestMergeQueue(t *testing.T) {
 	})
 
 	t.Run("non-collocated", func(t *testing.T) {
-		skip.WithIssue(t, 64056, "flaky test")
 		reset(t)
-		rangeID := rhs().RangeID
 		verifyUnmerged(t, store, lhsStartKey, rhsStartKey)
 		tc.AddVotersOrFatal(t, rhs().Desc().StartKey.AsRawKey(), tc.Target(1))
 		tc.TransferRangeLeaseOrFatal(t, *rhs().Desc(), tc.Target(1))
+		// NB: We're running on a 2 node `TestCluster` so we can count on the
+		// replica being removed to learn about the removal synchronously.
+		require.Equal(t, tc.NumServers(), 2)
 		tc.RemoveVotersOrFatal(t, rhs().Desc().StartKey.AsRawKey(), tc.Target(0))
-		testutils.SucceedsSoon(t, func() error {
-			_, err := tc.GetFirstStoreFromServer(t, 0).GetReplica(rangeID)
-			if err == nil {
-				return fmt.Errorf("replica still exists on dest %d", tc.Target(0))
-			} else if errors.HasType(err, (*roachpb.RangeNotFoundError)(nil)) {
-				return nil
-			}
-			return err
-		})
 
 		clearRange(t, lhsStartKey, rhsEndKey)
 		store.MustForceMergeScanAndProcess()

